camel/societies/role_playing.py

```diff
@@ -262,13 +278,22 @@ def init_agents(self, init_assistant_sys_msg: BaseMessage,
             output_language (str, optional): The language to be output by the
                 agents.
         """
+        if self.assist_funcs is not None:
+            assistant_config = FuncConfig(
+                functions=[func.as_dict() for func in self.assist_funcs],
+                function_call="auto")
+        else:
+            assistant_config = None
         self.assistant_agent = ChatAgent(
             init_assistant_sys_msg,
             self.model_type,
+            model_config=assistant_config,
             output_language=output_language,
+            function_list=self.assist_funcs,
             **(assistant_agent_kwargs or {}),
         )
         self.assistant_sys_msg = self.assistant_agent.system_message
```

Comment:
This is accessing the private field of a class. Please rework it into a property system_message or method get_system_message()

Reply:
Okay, it is not in the diff.